### PR TITLE
Added Javadocs to remaining classes.

### PR DESCRIPTION
**Pull request by PAndaContron**
Javadocs have been added to:

- BookcaseComponent
- BookComponent
- BookRecipeComponent
- DefaultDocumentData
- PageEditor
- RecipeParagraph

### CODE REVIEW

File: src/main/java/org/terasology/books/DefaultDocumentData.java [hunk 1]
```diff
@@ -24,28 +24,46 @@
...
+/**
+ * Holds the data for a page of a book, consisting of a list of paragraphs.
+ */
 public class DefaultDocumentData implements DocumentData {
     private DocumentRenderStyle documentRenderStyle;
     private List<ParagraphData> paragraphs = new LinkedList<>();
 
+    /**
+     * Creates a new instance with the specified render style.
+     */
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java:38]: You need to add `@param` annotations describing each parameter required by the method/constructor.

File: src/main/java/org/terasology/books/DefaultDocumentData.java [hunk 2]
```diff
@@ -24,28 +24,46 @@
...
+     * Gets an unmodifiable list of the paragraphs on this page.
+     */
     @Override
     public Collection<ParagraphData> getParagraphs() {
         return Collections.unmodifiableList(paragraphs);
     }
 
+    /**
+     * Adds another paragraph to this page.
+     */
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java:67]: Parameter description required

File: src/main/java/org/terasology/books/DefaultDocumentData.java [hunk 3]
```diff
@@ -24,28 +24,46 @@
...
     public void addParagraph(ParagraphData paragraph) {
         paragraphs.add(paragraph);
     }
 
+    /**
+     * Adds several paragraphs to this page.
+     */
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java:76]: Parameter description required

File: src/main/java/org/terasology/books/RecipeParagraph.java [hunk 4]
```diff
@@ -62,6 +75,9 @@ public RecipeParagraph(List<Block> blockIngredients, List<Prefab> itemIngredient
         resultIcon.setQuantity(resultCount);
     }
 
+    /**
+     * This paragraph's horizontal alignment will always be centered.
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java]: This description doesn't actually describe what the method does. You should describe the method as something that gets the paragraph render style. Then in a second sentence you could say something like "note that this paragraph's horizontal alignment will always be centered"

File: src/main/java/org/terasology/books/RecipeParagraph.java [hunk 5]
```diff
@@ -86,12 +109,23 @@ private void initializeForItem(ItemIcon itemIcon, Prefab itemIngredient) {
...
+     * @param blockIngredient The block to get the model of.
+     */
     private void initializeForBlock(ItemIcon itemIcon, Block blockIngredient) {
         itemIcon.setMesh(blockIngredient.getMesh());
         itemIcon.setMeshTexture(Assets.getTexture("engine:terrain").get());
         itemIcon.setTooltip(blockIngredient.getDisplayName());
     }
 
+    /**
+     * {@inheritDoc}
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java]: I would prefer if you didn't inheritDoc here. For most modules, we don't actually compile the Javadoc. The Javadocs are there mostly for people reading the code. So for someone reading this file, the inheritDoc could be a bit annoying (definitely not descriptive).

File: src/main/java/org/terasology/books/RecipeParagraph.java [hunk 6]
```diff
@@ -107,16 +141,30 @@ public void renderContents(Canvas canvas, Vector2i startPos, ContainerRenderSpac
         canvas.drawWidget(resultIcon, Rect2i.createFromMinAndSize(x, y, iconSize, iconSize));
     }
 
+    /**
+     * @param defaultStyle Not used
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:156]: Needs method description

File: src/main/java/org/terasology/books/RecipeParagraph.java [hunk 7]
```diff
@@ -107,16 +141,30 @@ public void renderContents(Canvas canvas, Vector2i startPos, ContainerRenderSpac
...
+     * @param containerRenderSpace Not used
+     * @param sideIndents Not used
+     */
     @Override
     public int getPreferredContentsHeight(ParagraphRenderStyle defaultStyle, int yStart, ContainerRenderSpace containerRenderSpace, int sideIndents) {
         return getPreferredSize().y;
     }
 
+    /**
+     * @param defaultStyle Not used
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:171]: Method description and `@return` missing

File: src/main/java/org/terasology/books/DefaultDocumentData.java [hunk 8]
```diff
@@ -24,28 +24,46 @@
...
+    /**
+     * Gets the render style of this page.
+     */
     @Override
     public DocumentRenderStyle getDocumentRenderStyle() {
         return documentRenderStyle;
     }
 
+    /**
+     * Gets an unmodifiable list of the paragraphs on this page.
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java]: Needs `@return` description

File: src/main/java/org/terasology/books/RecipeParagraph.java [hunk 9]
```diff
@@ -62,6 +75,9 @@ public RecipeParagraph(List<Block> blockIngredients, List<Prefab> itemIngredient
         resultIcon.setQuantity(resultCount);
     }
 
+    /**
+     * This paragraph's horizontal alignment will always be centered.
+     */
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:84]: Needs `@return`

File: src/main/java/org/terasology/books/RecipeParagraph.java [hunk 10]
```diff
@@ -107,16 +141,30 @@ public void renderContents(Canvas canvas, Vector2i startPos, ContainerRenderSpac
         canvas.drawWidget(resultIcon, Rect2i.createFromMinAndSize(x, y, iconSize, iconSize));
     }
 
+    /**
+     * @param defaultStyle Not used
+     * @param yStart Not used
+     * @param containerRenderSpace Not used
+     * @param sideIndents Not used
+     */
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:162]: Needs `@return`

File: src/main/java/org/terasology/books/logic/BookRecipeComponent.java [hunk 11]
```diff
@@ -23,12 +23,23 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * Used in a prefab to represent a recipe. The prefab can then be inserted in a book using a &ltrecipe&gt tag inside a page as such:
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/logic/BookRecipeComponent.java]: Please get rid of the `&lt` and `&gt`. They will probably work if the Javadocs are compiled, but they make for bad readability

File: src/main/java/org/terasology/books/logic/BookRecipeComponent.java [hunk 12]
```diff
@@ -23,12 +23,23 @@
...
+ * <br>
+ * <code>&ltrecipe module:prefab&gt</code>
```
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/logic/BookRecipeComponent.java]: Again, I would prefer if you got rid of the &lt/gt

**Review by mjuvekar7 — CHANGES_REQUESTED** [review]

**Comment by PAndaContron** [comment]
Ok, I will make sure to change these. Most of the tags are missing because I was under the impression that the Javadocs will  be compiled (I use Eclipse so it does that automatically on the side), so they would be inherited. I'll fix it when I get home.

**Comment by Waterpicker** [comment]
Approved after I dealt with a merge conflict.

**Pull request merged**